compiler/rustc_symbol_mangling/src/v0.rs

```diff
@@ -522,6 +520,10 @@ impl<'tcx> Printer<'tcx> for &mut SymbolMangler<'tcx> {
             }
 
             ty::GeneratorWitness(_) => bug!("symbol_names: unexpected `GeneratorWitness`"),
+
+            ty::Opaque(..) | ty::Projection(..) | ty::TyAlias(..) => {
+                bug!("symbol_names: unexpected projection")
+            }
```

Comment:
This is broken, please add a test (or I guess a new entry in an existing symbol mangling test) that does something like this (see [expected output](https://godbolt.org/z/vaq3a7z5M)):

```rust
pub trait Trait<T: ?Sized> {
    fn method() -> &'static ();
}

pub struct Foo<T>(T);
impl<T: std::ops::Deref> Trait<T::Target> for Foo<T> {
    fn method() -> &'static () {
        static FOO: () = ();
        &FOO
    }
}
```

(and I suppose it would also want to showcase a type alias with this PR)

The exact details of such mangling don't matter as much, as long as it produces some reasonable placeholder - the path is just fine.

Reply:
I added it as well so once everything is fixed, we will be able to ensure it's working as expected. Thanks for it!